src/accessibility/describe.js

```diff
@@ -0,0 +1,259 @@
+ * fill('red');
+ * ellipse(66.6, 66.6, 20, 20);
+ * ellipse(83.2, 66.6, 20, 20);
+ * triangle(91.2, 72.6, 75, 95, 58.6, 72.6);
+ * </code>
+ * </div>
+ */
+
+p5.prototype.describeElement = function(n, t, d) {
+  p5._validateParameters('describeElement', arguments);
+  const cnvId = this.canvas.id;
+  t = this._descriptionText(t);
+  n = this._elementName(n);
+  //Creates a sub DOM inside of the canvas with a table, populates
+  //a row header cell with the name of the elements and adds the description
+  //of the element in adjecent cell.
+  if (document.getElementById(cnvId + '_Description') === null) {
+    document.getElementById(cnvId).innerHTML =
+      '<div id="' +
+      cnvId +
+      '_Description" role="region" aria-label="Canvas Description"><table id="' +
+      cnvId +
+      '_eDsc"><caption>Canvas elements and their descriptions</caption></table></div>';
+  } else if (document.getElementById(cnvId + '_eDsc') === null) {
+    document
+      .getElementById(cnvId + '_dsc')
+      .insertAdjacentHTML(
+        'afterend',
+        '<table id="' +
+          cnvId +
+          '_eDsc"><caption>Canvas elements and their descriptions</caption></table>'
+      );
+  }
+  if (document.getElementById(cnvId + n) === null) {
+    let tr = document.createElement('tr');
+    tr.id = cnvId + n;
+    document.getElementById(cnvId + '_eDsc').appendChild(tr);
+  }
+  if (
+    document.getElementById(cnvId + n).innerHTML !==
+    '<th scope="row">' + n + '</th><td>' + t + '</td>'
+  ) {
+    document.getElementById(cnvId + n).innerHTML =
+      '<th scope="row">' + n + '</th><td>' + t + '</td>';
+  }
+  //If display is LABEL creates a div adjacent to the canvas element with
+  //a table, a row header cell with the name of the elements,
+  //and adds the description of the element in adjecent cell.
+  if (d === 'label') {
+    if (document.getElementById(cnvId + '_Label') === null) {
+      document
+        .getElementById(cnvId)
+        .insertAdjacentHTML(
+          'afterend',
+          '<div id="' +
+            cnvId +
+            '_Label" class="p5Label"><table id="' +
+            cnvId +
+            '_eLbl"></table></div>'
+        );
+    } else if (document.getElementById(cnvId + '_eLbl') === null) {
+      document
+        .getElementById(cnvId + '_dLbl')
+        .insertAdjacentHTML(
+          'afterend',
+          '<table id="' + cnvId + '_eLbl"></table>'
+        );
+    }
+    if (document.getElementById(cnvId + n + 'Lbl') === null) {
+      let tr = document.createElement('tr');
+      tr.id = cnvId + n + 'Lbl';
+      document.getElementById(cnvId + '_eLbl').appendChild(tr);
+    }
+
+    if (
+      document.getElementById(cnvId + n + 'Lbl').innerHTML !==
+      '<th scope="row">' + n + '</th><td>' + t + '</td>'
+    ) {
+      document.getElementById(cnvId + n + 'Lbl').innerHTML =
+        '<th scope="row">' + n + '</th><td>' + t + '</td>';
+    }
+  }
+};
+/**
+ * Helper function for describeElement().
```

Comment:
a short description of what updates are being made to the text would be helpful here